gen-windows.bat

```diff
@@ -1,5 +1,5 @@
 @echo off
 
-REM Generate project files for Visual Studio 2019
-call cmake -S . -B build-windows -G "Visual Studio 16 2019" -DCMAKE_TOOLCHAIN_FILE=./cmake/toolchain/windows-x64.cmake
+REM Generate project files for Visual Studio 2022
+call cmake -S . -B build-windows -G "Visual Studio 17 2022" -DCMAKE_TOOLCHAIN_FILE=./cmake/toolchain/windows-x64.cmake
```

Comment:
Recommend remove `-G "Visual Studio 17 2022"` and let whatever visual studio version compiler installed be used.

Reply:
Eu what do you said ? is obligatory for cake

Reply:
cmake can autodetect latest MSVC in the developer tools command prompt/powershell. there's no need to explicitly specify the generator.

Reply:
![image](https://user-images.githubusercontent.com/3403183/145713924-d8133f60-d977-4c9a-b9c2-8f62b593dcbf.png)

Reply:
> Recommend remove `-G "Visual Studio 17 2022"` and let whatever visual studio version compiler installed be used. Don't need to specify a specific VS version in my opinion.

`-G` argument is not for compilers but for [project generators](https://cmake.org/cmake/help/latest/manual/cmake-generators.7.html), and they aren't the same thing. Even if CMake would normally default to use MSBuild as the project generator on a VS Command Prompt, it also depends on the compilation environment. Since the environment is not even controlled by the devs but rather by GitHub, leaving an explicit declaration to the project generator is safer than not including it.

Reply:
i see, i have all time see dev said explicit version, and cmake gui also request so 🤔

Reply:
OK I wasn't aware this was being used as an artifact for GitHub Actions CI.

Reply:
small see, totaly forget, but after moving on github action, it no using anymore this gen-window, is only for Windows user/dev, for compile it only using indicztion inside workflow